src/collective/solr/solr.py

```diff
@@ -225,11 +251,24 @@ def add(self, boost_values=None, **fields):
         else:
             lst.append('<doc>')
         for f, v in fields.items():
+
+            # Add update="set" attribute to each field except for the uniqueKey
+            # field.
+            if f == uniqueKey:
+                tmpl = '<field name="%s">%%s</field>' % self.escapeKey(f)
+                lst.append(tmpl % self.escapeVal(v))
+                continue
+
             if f in boost_values:
-                tmpl = '<field name="%s" boost="%s">%%s</field>' % (
+                tmpl = '<field name="%s" boost="%s" update="set">%%s</field>' % (
```

Comment:
Hmm - since index time boosting doesn't work together with atomic updates, I think this can (should?) be handled differently here (no boosting if atomic updates enabled). But please double check, maybe also check with @buchi, I remember him having looked into that a bit more in depth.